src/dscanner/analysis/logic_precedence.d

```diff
@@ -19,41 +15,60 @@ import dsymbol.scope_;
  * if (a && (b || c)) // good
  * ---
  */
-final class LogicPrecedenceCheck : BaseAnalyzer
+extern(C++) class LogicPrecedenceCheck(AST) : BaseAnalyzerDmd!AST
 {
-	alias visit = BaseAnalyzer.visit;
-
 	enum string KEY = "dscanner.confusing.logical_precedence";
 	mixin AnalyzerInfo!"logical_precedence_check";
+	alias visit = BaseAnalyzerDmd!AST.visit;
 
-	this(string fileName, const(Scope)* sc, bool skipTests = false)
+	extern(D) this(string fileName, bool skipTests = false)
 	{
-		super(fileName, sc, skipTests);
+		super(fileName, skipTests);
 	}
 
-	override void visit(const OrOrExpression orOr)
+	override void visit(AST.LogicalExp le)
 	{
-		if (orOr.left is null || orOr.right is null)
-			return;
-		const AndAndExpression left = cast(AndAndExpression) orOr.left;
-		const AndAndExpression right = cast(AndAndExpression) orOr.right;
-		if (left is null && right is null)
-			return;
-		if ((left !is null && left.right is null) && (right !is null && right.right is null))
-			return;
-		addErrorMessage(orOr.line, orOr.column, KEY,
+		import dmd.tokens : EXP;
+
+		auto left = le.e1.isLogicalExp();
+		auto right = le.e2.isLogicalExp();
+
+		if (left)
+			left = left.op == EXP.andAnd ? left : null;
+		if (right)
+			right = right.op == EXP.andAnd ? right : null;
+
+		if (le.op != EXP.orOr)
+			goto END;
+
+		if (!left && !right)
+			goto END;
+		
+		if (left && left.parens)
+			goto END;
+
+		if (right && right.parens)
+			goto END;
```

Comment:
Why not put them in a single condition? We wasting lines of code this way.

Reply:
Addressed